ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java

```diff
@@ -1,74 +1,71 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
 package org.openmrs.reference;
 
+import org.hamcrest.Matchers;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
 import org.openmrs.reference.helper.PatientGenerator;
 import org.openmrs.reference.helper.TestPatient;
+import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
 import org.openmrs.reference.page.HeaderPage;
 import org.openmrs.reference.page.HomePage;
 import org.openmrs.reference.page.RegistrationPage;
 import org.openmrs.uitestframework.test.TestBase;
 
 import java.awt.*;
 
+import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
 
 
-public class DuplicatePatientRegisterTest  extends TestBase {
-    private HeaderPage headerPage;
-    private RegistrationPage registrationPage;
-    private HomePage homePage;
+public class DuplicatePatientRegisterTest  extends ReferenceApplicationTestBase {
     private TestPatient patient1;
 
     @Before
     public void setUp() throws AWTException {
-        headerPage = new HeaderPage(driver);
         patient1 = PatientGenerator.generateTestPatient();
-
-        assertPage(page);
-        
-        homePage = new HomePage(page);
-        registrationPage = new RegistrationPage(page);
-        assertPage(homePage);
     }
 
-    private void registerAPatient(TestPatient patient) throws InterruptedException {
-
-        homePage.goToRegisterPatientApp();
-        patient.familyName = "Smith";
-        patient.middleName = "";
-        patient.givenName = "Bob";
-        patient.gender = "Male";
-        patient.birthYear = "1999";
-        patient.birthMonth = "January";
-        patient.birthDay = "1";
-        patient.address1 = "address";
-        patient.address2 = "address";
-        registrationPage.enterPatient(patient);
+    // Test for RA-714
+    @Test
+    @Category(BuildTests.class)
+    public void duplicateRegisterTest() throws InterruptedException {
+        RegistrationPage registrationPage = homePage.goToRegisterPatientApp();
+        registrationPage.enterPatient(patient1);
+        ClinicianFacingPatientDashboardPage dashboardPage = registrationPage.confirmPatient();
+
+        patient1.uuid = dashboardPage.getPatientUuidFromUrl();
+
+        dashboardPage.goToPage("/referenceapplication/home.page");
```

Comment:
Let's actually move it to a new goToHomePage() method in a dashboardPage, which returns HomePage.